ultravox/tools/ds_tool.py

```diff
@@ -0,0 +1,172 @@
+import dataclasses
+import os
+from typing import Any, Dict, Optional, Union
+
+import datasets
+import openai
+import simple_parsing
+
+from ultravox.tools import tts
+
+CHAT_CLIENT = openai.Client()
+TTS_CLIENT = tts.AzureTts()
+
+DEFAULT_TEXTGEN_TEMPLATE = """Passage: {passage}
+
+Question: {question}
+
+Answer: {answer}
+
+Provide a short explanation to the question given the passage that entails the answer."""
+
+
+@dataclasses.dataclass(frozen=True)
+class TtsTask:
+    column_name: str = simple_parsing.field(default="question", alias="-c")
+    audio_column_name: Optional[str] = simple_parsing.field(default=None, alias="-a")
+    voice: Optional[str] = simple_parsing.field(default=None, alias="-V")
+    sample_rate: int = simple_parsing.field(default=16000, alias="-r")
+
+
+@dataclasses.dataclass(frozen=True)
+class TextGenerationTask:
+    language_model: str = simple_parsing.field(default="gpt-4o", alias="-m")
+    new_column_name: str = simple_parsing.field(default="explanation", alias="-c")
+    template: str = DEFAULT_TEXTGEN_TEMPLATE
```

Comment:
don't need to worry about this for now but I think we should allow @foo.txt syntax to load the template from a file (similar to curl @data.json)